From 478c7265b2693e5afe9d5db397ec4d8786c96b8f Mon Sep 17 00:00:00 2001
From: Vladyslav Andrishko <vladyslav.andrishko@globallogic.com>
Date: Wed, 4 Oct 2023 08:36:51 +0300
Subject: [PATCH 31/40] drm: rcar-rvgc: set the lacking DMA mask

The rcar-rvgc drm device lacks of the coherent_dma_mask definition,
so this patch adds it

However, the implementation differs from that in the Linux BSP.

Signed-off-by: Vladyslav Andrishko <vladyslav.andrishko@globallogic.com>
---
 drivers/gpu/drm/rcar-rvgc/rcar_rvgc_drv.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_drv.c b/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_drv.c
index 5e0309f352ad..1619f033b971 100644
--- a/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_drv.c
+++ b/drivers/gpu/drm/rcar-rvgc/rcar_rvgc_drv.c
@@ -10,6 +10,7 @@
 #include <linux/slab.h>
 #include <linux/wait.h>
 #include <linux/kthread.h>
+#include <linux/dma-mapping.h>
 
 #include <drm/drm_device.h>
 #include <drm/drm_atomic_helper.h>
@@ -235,6 +236,9 @@ static int rcar_rvgc_probe(struct rpmsg_device* rpdev) {
 	}
 
 	ddev->irq_enabled = 1;
+	ret = dma_coerce_mask_and_coherent(ddev->dev, DMA_BIT_MASK(32));
+	if (ret)
+		goto error;
 
 	/*
 	 * Register the DRM device with the core and the connectors with
-- 
2.25.1

